### PR TITLE
Add notifications support for Gitter

### PR DESCRIPTION
**Pull request by JafethDC**

### CODE REVIEW

File: lib/discourse_chat/provider/gitter/gitter_provider.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+module DiscourseChat
+  module Provider
+    module GitterProvider
+      PROVIDER_NAME = 'gitter'.freeze
+      PROVIDER_ENABLED_SETTING = :chat_integration_gitter_enabled
+      CHANNEL_PARAMETERS = [
+        { key: "name", regex: '^\S+$', unique: true },
+        { key: "webhook_url", regex: '^https://webhooks.gitter.im/e/\S+$', unique: true, hidden: true }
```
Comment (davidtaylorhq) [inline lib/discourse_chat/provider/gitter/gitter_provider.rb]: Very tiny thing here - a `.` in a regular expression allows any character. So technically

```
https://webhooksAgitterAim/e/blah
```
would match.

Solution is just to escape both dots with a backslash

**Comment by davidtaylorhq** [comment]
This looks great, thanks @JafethDC

**Comment by davidtaylorhq** [comment]
@JafethDC would you be able to write a how-to article on Meta with some screenshots showing how to get Gitter set up? Here's an existing one as an example: https://meta.discourse.org/t/set-up-hipchat-notifications-using-the-discourse-chat-integration-plugin/66734?u=david_taylor

**Comment by JafethDC** [comment]
Sure! I will do it.

**Pull request merged**